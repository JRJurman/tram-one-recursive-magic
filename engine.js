import buildApp from './buildApp'

export const getEngine = () => {
  if (typeof window === 'undefined') {
    return buildApp().webEngine
  }
  window.tramEngine = window.tramEngine || {}
  return window.tramEngine
}

// method for setting up the store for tests
export const scaffoldEngine = () => {
  window.tramEngine = buildApp().webEngine
}

export default getEngine
