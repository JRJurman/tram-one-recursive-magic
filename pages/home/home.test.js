import { scaffoldEngine } from '../../engine'
import homePage from './home'

describe('home page', () => {
  beforeEach(() => {
    // setup the store on the window
    scaffoldEngine()
  })
  test('snapshot', () => {
    const tree = homePage()
    expect(tree).toMatchSnapshot()
  })
})
