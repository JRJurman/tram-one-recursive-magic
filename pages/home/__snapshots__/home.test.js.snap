// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home page snapshot 1`] = `
<div>
  <h1
    style="
    color: #e6ef9f; background: black;
    cursor: pointer; text-align: center;
    padding: 1em; user-select: none;
    margin: -8px -8px 8px -8px;
  "
  >
    
      tram-one-example
    
  </h1>
  <noscript>
    <div
      style="
  border: solid;
  border-radius: 1em;
  padding: 0 1em;
  background: #fff0f0;
"
    >
      
        Oh No! It looks like JS has been disabled. Please enable it to use this applicaiton.
      
    </div>
  </noscript>
  <div>
    
        Thank you for using Tram-One!
    <br />
    
        To get started, edit 
    <code>
      tram-one-example/pages/home.js
    </code>
    .
      
  </div>
</div>
`;
